src/synchronization_c.c4

```diff
@@ -164,3 +304,100 @@ SHMEM_BIND_C_SYNC(`SHMEM_DEF_WAIT_UNTIL')
     }
 
 SHMEM_BIND_C_SYNC(`SHMEM_DEF_TEST')
+
+
+#define SHMEM_DEF_TEST_ALL(STYPE,TYPE)                                                         \
+    int SHMEM_FUNCTION_ATTRIBUTES                                                              \
+    shmemx_##STYPE##_test_all(TYPE *vars, size_t nelems, int cond, TYPE value)                 \
+    {                                                                                          \
+        int cmpret, ncompleted = 0;                                                            \
+        SHMEM_ERR_CHECK_INITIALIZED();                                                         \
+        SHMEM_ERR_CHECK_SYMMETRIC(vars, sizeof(TYPE));                                         \
+        SHMEM_ERR_CHECK_CMP_OP(cond);                                                          \
+                                                                                               \
+        if (nelems == 0) return 1;                                                             \
+                                                                                               \
+        for (int i = 0; i < nelems; i++) {                                                     \
+            COMP(cond, vars[i], value, cmpret);                                                \
+            if (cmpret) ncompleted++;                                                          \
+        }                                                                                      \
```

Comment:
Super minor comment, but I would put the declaration for `cmpret` into the loop body above.  That gives it the narrowest lexical scoping (and is a more defensive coding practice).